node/consensus/engine.go

```diff
@@ -339,7 +342,25 @@ func (ce *ConsensusEngine) Start(ctx context.Context, proposerBroadcaster Propos
 	ce.startMining(ctx)
 
 	// start the event loop
-	return ce.runConsensusEventLoop(ctx)
+	ce.wg.Add(1)
+	go ce.runConsensusEventLoop(ctx)
+
+	ce.wg.Wait()
+	ce.log.Info("Consensus engine stopped")
+	return nil
+}
+
+func (ce *ConsensusEngine) Close() {
```

Comment:
I think this should be done after the `ce.wg.Wait` in `Start`.  (1) Exported like this it looks like this is how caller stops CE, but it's actually by cancelling Start's context, (2) if `runConsensusEventLoop` returns early, it can/should cancel an context internal to `Start` to get it to pass Wait and thus hit close to clean up any tx.